api/src/main/java/com/qaengine/controllers/AnswerController.java

```diff
@@ -0,0 +1,92 @@
+package com.qaengine.controllers;
+
+import com.qaengine.database.AnswerRepository;
+import com.qaengine.database.QuestionRepository;
+import com.qaengine.exceptions.ResourceNotFoundException;
+import com.qaengine.lib.HelperFunctions;
+import com.qaengine.models.Answer;
+import com.qaengine.models.Question;
+import com.qaengine.models.inputs.AnswerInput;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@RestController
+public class AnswerController {
+    @Autowired
+    QuestionRepository questionRepository;
+    @Autowired
+    AnswerRepository answerRepository;
+    @Autowired
+    QuestionController questionController;
+
+    @CrossOrigin()
+    @PostMapping("questions/{questionId}/answers")
+    protected Answer addAnswer(
+            @PathVariable Long questionId,
+            @RequestBody @Valid AnswerInput answerinput
+    ) {
+        Question question = questionController.getQuestion(questionId);
+        Answer answer = new Answer(questionId);
+        HelperFunctions.copyProperties(answer, answerinput);
+        answer = answerRepository.save(answer);
+        question.getAnswers().add(answer);
+        questionRepository.save(question);
+        return answer;
+    }
+
+
+    @CrossOrigin()
+    @GetMapping("answers/{id}")
+    protected Answer getAnswer(@PathVariable Long id) {
+        Optional<Answer> answer = answerRepository.findById(id);
+        if (answer.isPresent()) {
+            return answer.get();
+        } else {
+            throw new ResourceNotFoundException();
+        }
+    }
+
+    @CrossOrigin()
+    @DeleteMapping("answers/{id}")
+    protected Long deleteAnswer(@PathVariable Long id) {
+        Answer answer = getAnswer(id);
```

Comment:
Same here, you shouldn't use controller logic inside another controller. 
Declare service layers for questions, answers and comments, put the isPresent() etc logic there, so you can reuse it from service layer